test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +92,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(testCommand.isMutating()) {
```

Comment:
Try to think about how you should test the else branch where the command is not mutating.